std/algorithm/iteration.d

```diff
@@ -2973,6 +2977,389 @@ template fold(fun...) if (fun.length >= 1)
     static assert(is(typeof(arr.fold!((a, b) => a)(1))));
 }
 
+/++
+Similar to $(LREF fold), but returns a range containing the successive reduced
+values. The call $(D cumulativeFold!(fun)(range, seed)) first assigns $(D seed)
+to an internal variable $(D result), also called the accumulator.
+The returned range contains the values $(D result = fun(result, x)) lazily
+evaluated for each element $(D x) in $(D range). Finally, the last element has
+the same value as $(D fold!(fun)(seed, range)).
+The one-argument version $(D cumulativeFold!(fun)(range)) works similarly, but
+it returns the first element unchanged and uses it as seed for the next
+elements.
+This function is also known as `partial_sum`, `accumulate`, `prefix_sum`,
+`scan`, or `cumulative_sum`.
+
+Returns:
+    a range containg the consecutive reduced values. If there is more than one
+    fun, the element type will be $(REF Tuple, std, typecons) containing one
+    element for each fun.
+
+See_Also:
+    $(WEB en.wikipedia.org/wiki/Prefix_sum, Prefix Sum)
++/
+template cumulativeFold(fun...)
+if (fun.length >= 1)
+{
+    import std.meta : staticMap;
+    alias binfuns = staticMap!(binaryFun, fun);
+
+    /++
```

Comment:
This is not part of the documentation, is it? Then you may want to clarify that in the comment shape.